pink/tasks/relative_frame_task.py

```diff
@@ -0,0 +1,238 @@
+    ) -> None:
+        r"""Set a new cost for all 3D orientation coordinates.
+
+        Args:
+            orientation_cost: Contribution of orientation errors to the
+                normalized cost, in :math:`[\mathrm{cost}] / [\mathrm{rad}]`.
+                If this is a vector, the cost is anisotropic and each
+                coordinate corresponds to an axis of the frame.
+        """
+        if isinstance(orientation_cost, float):
+            assert orientation_cost >= 0.0
+        else:  # not isinstance(orientation_cost, float)
+            assert all(cost >= 0.0 for cost in orientation_cost)
+        if isinstance(self.cost, np.ndarray):
+            self.cost[3:6] = orientation_cost
+        else:  # self.cost is not a vector
+            raise TaskDefinitionError(
+                "Frame task cost should be a vector, "
+                f"currently cost={self.cost}"
+            )
+
+    def set_target(
+        self,
+        transform_target_to_root: pin.SE3,
+    ) -> None:
+        """Set task target pose in the root frame.
+
+        Args:
+            transform_target_to_root: Affine transform matrix (4×4) from the
+                task target frame to the root frame.
+        """
+        self.transform_target_to_root = transform_target_to_root.copy()
+
+    def set_target_from_configuration(
+        self, configuration: Configuration
+    ) -> None:
+        """Set task target pose from a robot configuration.
+
+        Args:
+            configuration: Robot configuration.
+        """
+        self.set_target(configuration.get_transform(self.frame, self.root))
+
+    def compute_error(self, configuration: Configuration) -> np.ndarray:
+        r"""Compute the error twist.
+
+        The error is a twist :math:`e(q) \in se(3)` expressed in the local
+        frame (i.e., it is a *body* twist). We map it to :math:`\mathbb{R}^6`
+        following Pinocchio's convention where linear coordinates are followed
+        by angular coordinates. The error vector is:
+
+        .. math::
+
+            e(q) := {}_b \xi_{0b} = \log(T_{bt})
+
+        where :math:`b` denotes our frame, :math:`t` the target frame and
+        :math:`0` the inertial frame. See also :func:`Task.compute_error` for
+        more context.
+
+        Args:
+            configuration: Robot configuration :math:`q`.
+
+        Returns:
+            Frame task error :math:`e(q)`.
+        """
+        if self.transform_target_to_root is None:
+            raise TargetNotSet(
+                f"target pose of frame '{self.frame}' "
+                f"in frame '{self.root}' is undefined"
+            )
+        transform_frame_to_root = configuration.get_transform(
+            self.frame, self.root
+        )
+        transform_target_to_frame = transform_frame_to_root.actInv(
+            self.transform_target_to_root
+        )
+        error_in_frame: np.ndarray = pin.log(transform_target_to_frame).vector
+        return error_in_frame
```

Comment:
When it is one line of code, you can use the "Add suggestion" button (or <Ctrl+g>) on GitHub.

If it is more than one line of code, either the PR author does it, or you can commit directly to the PR (you are welcome to :smiley:). Or, in this instance since it was already merged, open a new PR :ok_hand: